TipCatDotNet.Api/Services/Auth/InvitationService.cs

```diff
@@ -105,33 +87,84 @@ public async Task<Result> Redeem(int memberId, CancellationToken cancellationTok
         }
 
 
-        public async Task<Result> Resend(int memberId, CancellationToken cancellationToken = default)
+        public Task<Result> Send(MemberContext memberContext, MemberRequest request, CancellationToken cancellationToken = default)
+        {
+            return Validate()
+                .Bind(EnsureInvitationExists)
+                .Bind(invitation => SendInternal(request, invitation, cancellationToken));
+
+
+            Result Validate()
+            {
+                var validator = new MemberRequestValidator(memberContext, _context);
+                return validator.ValidateInvite(request).ToResult();
+            }
+
+
+            async Task<Result<MemberInvitation>> EnsureInvitationExists()
```

Comment:
Should we move this method to validation layer?

Reply:
Done